include/mrdox/Metadata/Function.hpp

```diff
@@ -71,12 +71,39 @@ union FnFlags1
     BitField<6, 7, FunctionKind> functionKind;
 };
 
+// KRYSTIAN TODO: attributes (nodiscard, deprecated, and carries_dependency)
+// KRYSTIAN TODO: flag to indicate whether this is a function parameter pack
```

Comment:
If a function parameter list can have only one parameter pack, then maybe we should store the index of the pack in the `FunctionInfo` instead of putting a bool on each and every `Param` - to optimize for space.

However, if we have to add flags to parameters then the calculus changes, once we have one mandatory flag on a Param then we can add another bool for `isParameterPack` basically for free.